src/utils/daos_hdlr.c

```diff
@@ -1568,11 +1600,1359 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			D_GOTO(out, rc);
+		}
+	}
+	if (dst_file_dfs->type == DAOS) {
+		rc = dfs_umount(dst_file_dfs->dfs);
+		if (rc != 0) {
+			fprintf(stderr, "failed to unmount destination "
+				"(%d)\n", rc);
+			D_GOTO(out, rc);
+		}
+		rc = daos_cont_close(fa->dst_coh, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to close destination "
+				"container (%d)\n", rc);
+			D_GOTO(out, rc);
+		}
+		rc = daos_pool_disconnect(fa->dst_poh, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to disconnect from destination "
+				"pool "DF_UUIDF ": %s (%d)\n",
+				DP_UUID(fa->dst_p_uuid), d_errdesc(rc), rc);
+			D_GOTO(out, rc);
+		}
+	}
+out:
+	return rc;
+}
+
+static int fs_copy_parse_uuids(char *str,
+			       char **str_ptr,
+			       uuid_t *p_uuid,
+			       uuid_t *c_uuid,
+			       char **pool,
+			       char **cont)
+{
+	int rc = 0;
+
+	*pool = strtok_r(str, "/", str_ptr);
+	if (uuid_parse(*pool, *p_uuid) == 0) {
+		*cont = strtok_r(NULL, "/", str_ptr);
+		rc = uuid_parse(*cont, *c_uuid);
+		if (rc != 0) {
+			fprintf(stderr, "failed to parse cont uuid\n");
+			D_GOTO(out, rc);
+		}
+	} else {
+		fprintf(stderr, "failed to parse uuid or path\n");
+		D_GOTO(out, rc = EINVAL);
+	}
+out:
+	return rc;
+}
+
+static int
+fs_copy_parse(char *src,
+	      char *dst,
+	      struct fs_copy_args *fa,
+	      struct file_dfs *src_file_dfs,
+	      struct file_dfs *dst_file_dfs,
+	      struct duns_attr_t *src_dattr,
+	      struct duns_attr_t *dst_dattr,
+	      char **src_str,
+	      char **src_dfs_path,
+	      int *src_str_len,
+	      char **dst_str,
+	      char **dst_dfs_path,
+	      int *dst_str_len)
+{
+	/* parse src and dst strings
+	 * for DAOS, daos: specified in string
+	 * for POSIX, just use path
+	 * example: --src={daos://pool/cont}
+	 * example: --dst={/a/path}
+	 */
+	int	rc = 0;
+	char	*src_saveptr = NULL;
+	char	*dst_saveptr = NULL;
+	char	*src_type = NULL;
+	char	*dst_type = NULL;
+	char	*src_pool = NULL;
+	char	*src_cont = NULL;
+	char	*dst_pool = NULL;
+	char	*dst_cont = NULL;
+
+	/* validate and enforce "daos://" prefix is used, if not,
+	 * set to "path" type
+	 */
+	if (strncmp(src, "daos://", 7) == 0) {
+		src_type = "daos";
+		/* start of src pool/cont str */
+		src += 7;
+		/* daos:// should not have more than two
+		 * forward slashes (i.e. daos:///)
+		 */
+		if (strncmp(src, "/", 1) == 0) {
+			fprintf(stderr, "cannot parse daos src type "
+				"format, please use:\n"
+				"\t--src=daos://<pool/cont>\n");
+			D_GOTO(out, rc = EINVAL);
+		}
+	} else {
+		src_type = "path";
+	}
+	if (strncmp(dst, "daos://", 7) == 0) {
+		dst_type = "daos";
+		/* start of dst pool/cont str */
+		dst += 7;
+		if (strncmp(dst, "/", 1) == 0) {
+			fprintf(stderr, "cannot parse daos dst type format, "
+				"please use:\n"
+				"\t--dst=daos://<pool/cont>\n");
+			D_GOTO(out, rc = EINVAL);
+		}
+	} else {
+		dst_type = "path";
+	}
+
+	/* check for src DAOS pool/cont or UNS path */
+	if (strcmp(src_type, "daos") == 0) {
```

Comment:
@dsikich I wonder if we should abstract this parsing to a function, to avoid essentially duplicating the logic for parsing the src and dst?

Reply:
we should be using the gurt macros (D_*) for str functions; for correct memory tracing, but im ok that being done in a different PR. please keep track of this in a ticket or whatever.

Reply:
@mchaarawi I do not see a macro for strcmp in gurt/common.h? Only for strndup (D_STRNDUP)